File: system/monolith/dotnet/Core/Dtos/ViewOrderDetailsResponse.cs
using Optivem.EShop.Monolith.Core.Entities;

namespace Optivem.EShop.Monolith.Core.Dtos;

public class ViewOrderDetailsResponse
{
    public string OrderNumber { get; set; } = null!;
    public DateTime OrderTimestamp { get; set; }
    public string Sku { get; set; } = null!;
    public int Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal TotalPrice { get; set; }
    public OrderStatus Status { get; set; }
}

File: system/monolith/java/src/main/java/com/optivem/eshop/monolith/core/dtos/ViewOrderDetailsResponse.java
package com.optivem.eshop.monolith.core.dtos;

import com.optivem.eshop.monolith.core.entities.OrderStatus;
import lombok.Data;

import java.math.BigDecimal;
import java.time.Instant;

@Data
public class ViewOrderDetailsResponse {
    private String orderNumber;
    private Instant orderTimestamp;
    private String sku;
    private int quantity;
    private BigDecimal unitPrice;
    private BigDecimal totalPrice;
    private OrderStatus status;
}
